File: Problems/858_Mirror_Reflection.py
class Solution:
    def mirrorReflection(self, p: int, q: int) -> int:
        
        def gcd(m, n):
            if n == 0:
                return m
            return gcd(n, m%n)
        
        f = gcd(p, q)
        m = p*q//f
        if (m//q) % 2 == 0:
            return 2
        else:
            if (m // p) % 2== 1:
                return 1
            else:
                return 0

File: Problems/858_Mirror_Reflection.cpp
class Solution {
public:
    int mirrorReflection(int p, int q) {
        int f = gcd(p, q);
        int m = p*q/f;
        if ((m/q) % 2 == 0)
            return 2;
        if ((m / p) % 2 == 1)
            return 1;
        return 0;
    }
    int gcd(int m, int n) {
        if (n == 0) return m;
        return gcd(n, m%n);
    }
};